server/routes/barcode/PdfController.ts

```diff
@@ -36,4 +40,34 @@ export default class PdfController {
     const view = new PdfControllerView(req.session.pdfForm, req.flash('errors'))
     return res.render('pages/barcode/pdf/print-coversheets', { ...view.renderArgs })
   }
+
+  async submitPrintCoverSheet(req: Request, res: Response): Promise<void> {
+    if (!req.session.recipients) {
+      return res.redirect('/barcode/find-recipient')
+    }
+    if (!req.session.pdfForm) {
+      return res.redirect('/barcode/pdf/select-envelope-size')
+    }
+
+    const barcodeImages = await Promise.all(
+      req.session.recipients.map(async recipient => {
+        try {
+          const barcodeData = await this.createBarcodeService.generateBarcode(
+            req.session.createBarcodeAuthToken,
+            recipient
+          )
+          return barcodeData.barcodeImageDataUrl
+        } catch (error) {
+          logger.error(`Could not generate a barcode for ${recipient.prisonNumber}, ${error}`)
+          return undefined
+        }
+      })
+    )
+
+    return res.renderPDF('pdf/barcode-cover-sheet', {
+      envelopeSize: req.session.pdfForm.envelopeSize,
+      barcodeImages,
+      printDebugInfo: true,
```

Comment:
Need to make `printDebugInfo` a config param / env var .... but it's useful being on for the time being ...